mutex.go

```diff
@@ -44,10 +44,21 @@ var (
 // Lock waits up to duration d for l.Lock() to succeed.
 func Lock(l sync.Locker, d time.Duration) error {
 	done := make(chan struct{}, 1)
-	go func() {
-		l.Lock()
-		done <- struct{}{}
-	}()
+	m, ok := l.(*mutex)
+	// if l is a mutex, lock with a timeout context.
+	if ok {
+		ctx, cancel := context.WithTimeout(context.Background(), d*time.Second)
```

Comment:
I think github now offers this, similar to gerrit: you type in new comments and then submit them all at once. But, I have it say, the experience is still quite horrible :) I can see your new changes passed last reviewed commit but I'm missing all my comments. So, a reviewer either need to keep 2 tabs open or switch back and forth... not ideal at all.